anthos/cluster/bundled-lb-install-script.sh

```diff
@@ -1,25 +1,55 @@
 #!/bin/bash
+
+# print a message to stderr, prefixed by HOSTNAME
+function note() {
+  echo 1>&2 "$HOSTNAME: $*"
+}
+
+# print the given command to stderr, run it, and exit verbosely if it fails.
+function xrun() {
+  note "+ $@"
+  "$@" && return 0
+  local xstat=$?
+  note "Cmd $1 failed, exit $xstat"
+  exit "$xstat"
+}
+
+# fetch the pem from the given address, and print to stdout.
+fetch_pem() {
+  local addr=$1
+  xrun openssl s_client -showcerts -verify 5 -connect ${vcenter_fqdn}:443 < /dev/null \
+  | awk '/BEGIN/,/END/ {print}'
+  return "$((PIPESTATUS[0]))"
```

Comment:
What is the purpose of returning a string "0" here (rather than a 0 return code)? The result doesn't seem to be getting used anywhere.

Reply:
> What is the purpose of returning a string "0" here (rather than a 0 return code)?

in bash, double-quotes do not designate a string type, rather they are used to suppress post-expansion side effects such as word splitting.  it's considered a good practice to double-quote all variable references unless the intention is to allow word-splitting on the expanded result.

> The result doesn't seem to be getting used anywhere.

i wrote the function in a way that allows the return value to be used, even though the script currently doesn't check it.

Reply:
👍  I see. I now realize that I haven't used `return n` in a bash function before, I learned something new today. 

I've always written things in a way that the last command provided the return code, but here that command is replaced by awk.